primitives/src/vid/advz.rs

```diff
@@ -190,7 +190,8 @@ where
     #[serde(with = "canonical")]
     all_evals_digest: KzgEvalsMerkleTreeNode<E, H>,
 
-    bytes_len: usize, // TODO don't use usize in serializable struct?
+    bytes_len: usize,         // TODO don't use usize in serializable struct?
+    num_storage_nodes: usize, // TODO don't use usize in serializable struct?
```

Comment:
I wonder if we could infer either/both of these from the number of polynomial commitments and number of evals (height and width of matrix?)

Reply:
And yes, best to avoid `usize` in serialized structs. I don't think we're at the point where we're fighting for bytes in a succinct fixed-size type like this, `u64` is probably fine. Although in practice, both of these will probably never be bigger than a `u32`

Reply:
> I wonder if we could infer either/both of these from the number of polynomial commitments and number of evals (height and width of matrix?)

Possible. Number of polynomial commitments is readily available in `Common`. Number of evals... is this info readily available to all nodes?

Note that in the near future we will allow number of polynomials to decouple from number of evals for optimization reasons. Of course, the relevant ratio should be known to all but the calculation gets a bit more obfuscated.

Reply:
> Number of evals... is this info readily available to all nodes?

Yes. In hotshot, at least, it equals the number of quorum nodes. So everybody knows it.

Reply:
But note that the number of evals is not readily available from a `Common` or a single `Share`. So it's up to the downstream user to do the accounting required for this.

Reply:
Ah ok, we don't want this